frontend/src/components/TimerPanel/TimerPanel.tsx

```diff
@@ -0,0 +1,215 @@
+  startDate: string
+  endDate?: string
+  daysBetween: number
+  skipWeekends: boolean
+  skipHolidays: boolean
+  timeOfDay: string
+}
+const TimerPanel = () => {
+  // eslint-disable-next-line
+  const today: string = dayjs().format("YYYY-MM-DD")
+  // eslint-disable-next-line
+  const tomorrow: string = dayjs().add(1, "day").format("YYYY-MM-DD")
+  // eslint-disable-next-line
+  const aMonthFromToday: string = dayjs().add(1, "month").format("YYYY-MM-DD")
+
+  // eslint-disable-next-line
+  const { handleSubmit, register, control, watch, setValue } = useForm({
+    defaultValues: {
+      startDate: tomorrow,
+      endDate: aMonthFromToday,
+      daysBetween: 1,
+      timeOfDay: "08:00",
+      skipHolidays: false,
+      skipWeekends: false,
+    },
+  })
+
+  // eslint-disable-next-line
+  const endDate = watch("endDate")
+  // eslint-disable-next-line
+  const startDate = watch("startDate")
+
+  const showToast = useCustomToast()
+  const queryClient = useQueryClient()
+  const mutation = useMutation({
+    mutationFn: (data: TDataCreateSchedule) => createSchedule(data),
+    onSuccess: () => {
+      showToast("Success!", "Schedule created successfully.", "success")
+    },
+    onError: (err: ApiError) => {
+      handleError(err, showToast)
+    },
+    onSettled: async () => {
+      await queryClient.invalidateQueries({ queryKey: ["schedule"] })
+    },
+  })
+
+  const onSubmit: SubmitHandler<TimerFormData> = (data: TimerFormData) => {
+    const {
+      startDate,
+      endDate,
+      daysBetween,
+      skipHolidays,
+      skipWeekends,
+      timeOfDay,
+    } = data
+    const submissionData: {
+      schedule: {
+        skipHolidays: boolean
+        endDate: string | undefined
+        skipWeekends: boolean
+        daysBetween: number
+        timesOfDay: string[]
+        startDate: string
+      }
+    } = {
+      schedule: {
+        startDate,
+        endDate,
+        daysBetween,
+        skipHolidays,
+        skipWeekends,
+        timesOfDay: [timeOfDay],
+      },
+    }
+    mutation.mutate(submissionData)
+  }
+
+  return (
+    <Card variant="outline" w="80%">
+      <CardHeader bg="gray.100">
+        <Heading size="lg">Question Schedule</Heading>
+      </CardHeader>
+      <CardBody>
+        <form onSubmit={/* eslint-disable */ handleSubmit(onSubmit)}>
+          <Flex direction="column">
+            <FormRow>
+              <FormControl>
+                <FormLabel>Date Start</FormLabel>
+                <Controller
+                  name="startDate"
+                  control={control}
+                  rules={{ required: true }}
+                  render={({ field }) => (
+                    <Input
+                      {...field}
+                      type="date"
+                      min={today}
+                      onChange={(e) => {
+                        const newStartDate = e.target.value
+                        if (dayjs(newStartDate).isAfter(dayjs(endDate))) {
```

Comment:
now this is a little closer to actual business logic. (i.e. start date can't be after end date, and what to do if that happens.) In MVVM you'd have StartDate/EndDate properties that the display is bound to. When one of these changes, it does the business logic check, then if necessary updates the other. The displays then auto-update through the binding. 

The really nice thing about that separation is it's trivial to unit test.

Reply:
btw I'm not necessarily saying you need to change this implementation -- just wondering if a similar pattern is applicable to React or not. We should definitely be implementing industry standards; if this implementation is it, then disregard. @Sergei-S-V again FYI

Reply:
this is how it's done in react.  there's a few things going on behind the scenes, though.  the startdate/enddate are indeed part of the model -- the model here being the state of react application, which is being controlled by the react-form-hook.  that's why these particular components are surrounded by the tag `<Control>`